app/src/main/java/ch/epfl/sdp/MapsActivity.java

```diff
@@ -0,0 +1,48 @@
+package ch.epfl.sdp;
+
+import android.app.Activity;
+import android.content.Context;
+import android.location.LocationManager;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+
+import androidx.fragment.app.FragmentActivity;
+
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.SupportMapFragment;
+
+public class MapsActivity extends FragmentActivity implements OnMapReadyCallback {
+    private MapApi mapApi;
+    private Activity activity = this;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_maps);
+
+        Button button = findViewById(R.id.update_loc);
+        button.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                mapApi.updatePosition(activity);
+            }
+        });
+
+        mapApi = new GoogleApi((LocationManager) getSystemService(Context.LOCATION_SERVICE));
+
+        SupportMapFragment mapFragment = (SupportMapFragment) getSupportFragmentManager().findFragmentById(R.id.map);
+        mapFragment.getMapAsync(MapsActivity.this);
+    }
+
+    @Override
+    public void onMapReady(GoogleMap googleMap) {
+        ((GoogleApi) mapApi).setMap(googleMap);
+    }
+
+    public MapApi getMapApi() {

```

Comment:
This is bad practice, you should not expose internal fields just for texting purposes.

Instead, think about Dependency Injection to provide mock implementations to write tests.